test/Unit/Api/ClientTest.php

```diff
@@ -0,0 +1,548 @@
+    {
+        $body = array(rand() => rand());
+        $this->connection->expects($this->once())
+             ->method('get')
+             ->with('/store')
+             ->will($this->returnValue($body));
+
+        $this->assertSame($body, Client::getStore());
+    }
+
+    public function testGetRequestsRemainingReturnsTheValueFromTheLastHeader()
+    {
+        $this->connection->expects($this->once())
+             ->method('getHeader')
+             ->will($this->returnValue('12345'));
+
+        $this->assertSame(12345, Client::getRequestsRemaining());
+    }
+
+    public function testGetRequestsRemainingRequestsTimeWhenNoValueAvailable()
+    {
+        $this->connection->expects($this->exactly(2))
+             ->method('getHeader')
+             ->will($this->onConsecutiveCalls(false, '12345'));
+
+        $this->connection->expects($this->once())
+             ->method('get')
+             ->with('/time', false)
+             ->will($this->returnValue((object)array('time' => time())));
+
+        $this->assertSame(12345, Client::getRequestsRemaining());
+    }
+
+    public function collections()
+    {
+        return array(
+            //      path           function             classname
+            array('products',      'getProducts',      'Product'),
+            array('brands',        'getBrands',        'Brand'),
+            array('orders',        'getOrders',        'Order'),
+            array('customers',     'getCustomers',     'Customer'),
+            array('coupons',       'getCoupons',       'Coupon'),
+            array('orderstatuses', 'getOrderStatuses', 'OrderStatus'),
+            array('categories',    'getCategories',    'Category'),
+            array('options',       'getOptions',       'Option'),
+            array('optionsets',    'getOptionSets',    'OptionSet'),
+            array('products/skus', 'getSkus',          'Sku'),
+            array('requestlogs',   'getRequestLogs',   'RequestLog'),
+        );
+    }
+
+    /**
+     * @dataProvider collections
+     */
+    public function testGettingASpecificResourceReturnsACollectionOfThatResource($path, $fnName, $class)
+    {
+        $this->connection->expects($this->once())
+             ->method('get')
+             ->with('/' . $path, false)
+             ->will($this->returnValue(array(array(), array())));
+
+        $collection = Client::$fnName();
+        $this->assertInternalType('array', $collection);
+        foreach ($collection as $resource) {
+            $this->assertInstanceOf('Bigcommerce\\Api\\Resources\\' . $class, $resource);
+        }
+    }
+
+    /**
+     * @dataProvider collections
+     */
+    public function testGettingTheCountOfACollectionReturnsThatCollectionsCount($path, $fnName, $class)
+    {
+        if (in_array($path, array('coupons', 'orderstatuses', 'products/skus', 'requestlogs'))) {
+            $this->markTestSkipped(sprintf('Getting the count of %s is not supported', $path));
```

Comment:
I suppose a better message would be `The php api client doesn't support getting the count of %s`. I didn't check if the actual API support was there--merely that the client exposes that functionality.